### PR TITLE
Match scratch-www studios differently

### PR DESCRIPTION
**Pull request by WorldLanguages**
This only changes content script code
Behavior does not change in any way if URL doesn't match studio regex
If the URL matches a studio (and it didn't 404), `onInfoAvailable` waits until `document.head` is available before doing _anything_. This may add a slight delay before injecting userscripts and userstyles, even before Scratch transitions studios to scratch-www, but it's barely noticeable, if at all.
After the head is available, it checks if `addon.tab.clientVersion` would return `"scratch-www"`. If so, similarly to #2416 it will set the pseudoUrl of the page to `/studios_www/...` instead of `/studios/...`. Similarly to #2416, userscripts and userstyles will take a bit more time to get injected after studios become scratch-www (and now it can be probably seen by eye, specially with website dark mode). This isn't ideal but good enough to have a clean transition (of course, after the transition by Scratch, we can remove all of this and we won't have any other timing problems like this)

TODOs:
- [x] Adapt addons so that they can use this (cc @mxmou, feel free to push into this PR)
- [ ] Change regexes to consider `/studios_www/...` to be a scratch-www page

### CODE REVIEW

File: content-scripts/cs.js [hunk 1]
```diff
@@ -223,24 +223,37 @@ function setCssVariables(addonSettings, addonsWithUserstyles) {
   }
 }
 
+function waitForDocumentHead() {
+  if (document.head) return Promise.resolve();
+  else {
+    return new Promise((resolve) => {
+      const observer = new MutationObserver(() => {
+        if (document.head) resolve();
```
Comment (apple502j) [inline content-scripts/cs.js:231]: disconnect!

**Comment by apple502j** [comment]
I think this is kinda overkill.

**Comment by mxmou** [comment]
@apple502j It's better than using complex selectors to prevent CSS conflicts. `scratchr2.css` and `scratchwww.css` are separate because running both on all pages causes bugs.

**Comment by Hans5958** [comment]
what a bodge

**Comment by WorldLanguages** [comment]
The art of the bodge

**Comment by RedGuy12** [comment]
>  Adapt addons so that they can use this (cc @mxmou, feel free to push into this PR)
apple did that in #2493

**Comment by WorldLanguages** [comment]
> apple did that in #2493

I meant doing it for themes, not addons that use JS

**Comment by WorldLanguages** [comment]
> * [ ]  Change regexes to consider `/studios_www/...` to be a scratch-www page

Note: this TODO wasn't done. But no addon uses `scratchWWWNoProject` or `isNotScratchWWW` matches, so it doesn't matter

**Pull request merged**